### ISSUE TITLE
False positive `no-member` on args of an exception

### ISSUE THREAD

**Issue by mymedia2** [PyCQA/pylint#7233]
### Bug description

The following sample script triggers false positive 'no-member' error, although the code runs okay. Pylint considers the first element of the .args tuple as a str instance.

```python
from dataclasses import dataclass


class MyHelloError(RuntimeError):
    @property
    def detail(self):
        return self.args[0]


@dataclass
class ErrorInfo:
    errorcode: int


try:
    raise MyHelloError(ErrorInfo(1234))
except MyHelloError as exc:
    print(exc.detail.errorcode)
```

### Command used

```shell
pylint -d all -e E hello1.py
```

### Pylint output

```shell
************* Module hello1
hello1.py:18:10: E1101: Instance of 'str' has no 'errorcode' member (no-member)

-----------------------------------
Your code has been rated at 5.00/10
```

### Expected behavior

No error on correct code.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]
```

### OS / Environment

Ubuntu 22.04 (Jammy Jellyfish)

**Comment by DanielNoord** [comment]
Superseded by https://github.com/PyCQA/astroid/pull/1749.

**Comment by DanielNoord** [comment]
Fixed in https://github.com/PyCQA/astroid/pull/1749 which also includes the necessary regressions tests. We're nearing a new `pylint` release so this should be available relatively soon! 😄
